### PR TITLE
WOR-1510 upgrade otel override HttpServerMetrics

### PR DESCRIPTION
**Pull request by dvoet**
https://broadworkbench.atlassian.net/browse/WOR-1510

### CODE REVIEW

File: src/main/java/bio/terra/common/opentelemetry/HttpMetricsAdvice.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                if (d < 2) {
+                  return d + 0.1;
+                }
+                if (d < 10) {
+                  return d + 1;
+                }
+                if (d < 60) {
+                  return d + 5;
+                }
+                return d + 10;
```
Comment (tlangs) [inline src/main/java/bio/terra/common/opentelemetry/HttpMetricsAdvice.java:43]: Since this is Java 17, this could be a switch with guards.

Comment (dvoet) [reply src/main/java/bio/terra/common/opentelemetry/HttpMetricsAdvice.java:43]: copilot wrote this code :)

Comment (tlangs) [reply src/main/java/bio/terra/common/opentelemetry/HttpMetricsAdvice.java:43]: 🤖 🚫

Comment (dvoet) [reply src/main/java/bio/terra/common/opentelemetry/HttpMetricsAdvice.java:43]: I think guards is a preview feature of java 17 and we don't have that enabled

**Review by okotsopoulos — APPROVED** [review]
Looks good as far as I can tell.

Does a service using OTEL via TCL need to do anything extra to get these changes?  Is there anywhere I can see these updated metrics in action?

File: src/main/java/bio/terra/common/opentelemetry/HttpMetricsAdvice.java (hunk 1 above)
Comment (tlangs) [reply src/main/java/bio/terra/common/opentelemetry/HttpMetricsAdvice.java:43]: ah, darn

**Review by tlangs — APPROVED** [review]

**Comment by dvoet** [comment]
> Does a service using OTEL via TCL need to do anything extra to get these changes?

yes, they need to upgrade their TCL version, I will start doing that tomorrow

**Pull request merged**